### ISSUE TITLE
Integrate PNG

### ISSUE THREAD

**Issue by jakearchibald** [GoogleChromeLabs/squoosh#79]
1. [x] WASM PNG Quant #8
1. [ ] Integrate PNG Quant

After that, PNG has a slightly different flow to the other encoders, as it has 'crushers', and sometimes there's a benefit of passing the image through multiple crushers.

1. [ ] WASM two crushers #7 #9
1. [ ] Integrate 'crush' as a checkbox to PNG encoders, which passes the output through PNG crushers.

The main codec may be "Browser PNG" or "PNG Quant" (#8), but then zero or more crushers (#7 #9) can be applied afterwards. 

We need to investigate whether the crushers can be run in parallel, or whether there's a benefit to running them in series. I have a feeling that Zopfli PNG is best run at the end, after other crushers. I guess this depends on which crushers work on PNG data, and which work on raw image data.

1. [ ] Stretch goal: Allow each crusher to be enabled/disabled.
1. [ ] Stretch goal: Integrate options for each crusher.
1. [ ] Stretch goal: Show savings provided by each crusher.

Visualising this in the UI is interesting. It'd be nice to show the savings from the main codec, followed by the savings from each crusher. The crushers themselves have options, which dictate how much time they'll spend crushing the PNG.

**Comment by surma** [comment]
I’d vote for punting that to v2. For v1 we should definitely ship a PNG encoder, but maybe with a predefined set of crushers? If we have time left before CDS, we can pull that into v1, but I am not sure the architecture and UI can be done in time. My 2¢

**Comment by jakearchibald** [comment]
@surma I've edited the OP. Does this sound more reasonable?

**Comment by surma** [comment]
sg!

**Comment by jakearchibald** [comment]
I'll split this into seperate issues
